Makefile

```diff
@@ -127,6 +127,18 @@ vet:
 
 cover: depend
 	go test -tags=unit $(shell go list ./...) -cover
+	@rm -rf coverage.txt
+	@for d in `go list ./... | grep -v vendor`; do \
+		t=$$(date +%s); \
+		go test -coverprofile=cover.out -covermode=atomic $$d || exit 1; \
+		echo "Coverage test $$d took $$(($$(date +%s)-t)) seconds"; \
+		if [ -f cover.out ]; then \
+			cat cover.out >> coverage.txt; \
+			rm cover.out; \
+		fi; \
+	done
+	@echo "Uploading coverage results..."
+	@curl -s https://codecov.io/bash | bash
```

Comment:
Are we sure we want to do this? Piping untrusted code to bash seems dangerous to me :P